.github/workflows/docker.yaml

```diff
@@ -0,0 +1,65 @@
+name: Docker
+on:
+  push:
+    branches:
+      - main
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: 'Tag to use (defaults to "test")'
+        default: "test"
+
+permissions:
+  id-token: write
+  contents: read
+
+env:
+  VERSION: 2.0.0
+
+jobs:
+  docker:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash -l {0}
+    env:
+      DOCKER_BUILDKIT: 1
+      BUILDKIT_PROGRESS: plain
+      CLOUDSDK_CORE_DISABLE_PROMPTS: 1
+      IMAGE_NAME: clinvarbitration
+      DOCKER_DEV: australia-southeast1-docker.pkg.dev/cpg-common/images-tmp
+      DOCKER_MAIN: australia-southeast1-docker.pkg.dev/cpg-common/images
+    steps:
+      - uses: actions/checkout@v4
+
+      - id: "google-cloud-auth"
+        name: "Authenticate to Google Cloud"
+        uses: google-github-actions/auth@v2
+        with:
+          workload_identity_provider: "???"
```

Comment:
A note for @populationgenomics/software-team to chat about what we'd like to do here. 

We'll need to discuss how we enforce the environment stuff with pipelines. 
@nevoodoo and @violetbrina have had discussions about shifting this to cpg-infra. Do we want to do this? 
Or @nevoodoo can just quickly manually set up the workload identity providers for relevant pipelines.... 

Let's chat in stand up!

Reply:
Excellent point. I am definitely in favour of both using the workload_identity_provider and chatting in standup about it :)